src/tasks/AppleAppBuilder/Templates/runtime.m

```diff
@@ -201,6 +204,52 @@
 //%DllMap%
 }
 
+int32_t GlobalizationNative_LoadICUData(void *pData);
+
+static int32_t load_icu_data ()
+{
+    char path [1024];
+    int res;
+
+    const char *dname = "icudt.dat";
+    const char *bundle = get_bundle_path ();
+
+    os_log_info (OS_LOG_DEFAULT, "Loading ICU data file '%s'.", dname);
+    res = snprintf (path, sizeof (path) - 1, "%s/%s", bundle, dname);
+    assert (res > 0);
+
+    FILE *fp = fopen (path, "rb");
+    if (fp != NULL) {
+        if (fseek (fp, 0L, SEEK_END) != 0) {
+            os_log_info (OS_LOG_DEFAULT, "Unable to determine size of the dat file");
+            exit(1);
+        }
+
+        long bufsize = ftell (fp);
+
+        icu_data = malloc (sizeof (char) * (bufsize + 1));
+
+        if (fseek (fp, 0L, SEEK_SET) != 0) {
+            os_log_info (OS_LOG_DEFAULT, "Unable to seek ICU dat file.");
+            exit(1);
+        }
+
+        size_t len = fread (icu_data, sizeof (char), bufsize, fp);
+        if ( ferror ( fp ) != 0 ) {
+            os_log_info (OS_LOG_DEFAULT, "Unable to read ICU dat file");
+            exit (1);
+        }
+
+        fclose (fp);
+    }
+    else {
```

Comment:
I would get rid of the larger conditional if we're just going to exit out when the file doesn't exit. We should just follow the same pattern used for all the calls above of checking with a short conditional immediately after the call.